GAE/src/org/waterforpeople/mapping/dataexport/service/BulkDataServiceClient.java

```diff
@@ -989,24 +989,25 @@ public static DeviceFilesDto parseDeviceFile(JSONObject json)
                                     }
                                     dto.setOptionContainerDto(container);
                                 }
+                            }
 
-                                // The questionDependency check below is related to the
-                                // previous if statements dependentFlag, dependentQuestionId,
-                                // dependentQuestionAnswer i.e. checks whether question is
-                                // dependent on another
-                                if (json.has("questionDependency")
-                                        && !JSONObject.NULL.equals(json
-                                                .get("questionDependency"))) {
-                                    QuestionDependencyDto dep = new QuestionDependencyDto();
-                                    JSONObject depJson = json
-                                            .getJSONObject("questionDependency");
-                                    dep.setQuestionId(depJson
-                                            .getLong("questionId"));
-                                    dep.setAnswerValue(depJson
-                                            .getString("answerValue"));
-                                    dto.setQuestionDependency(dep);
-                                }
+                            // The questionDependency check below is related to the
+                            // previous if statements dependentFlag, dependentQuestionId,
+                            // dependentQuestionAnswer i.e. checks whether question is
+                            // dependent on another
+                            if (json.has("questionDependency")
+                                    && !JSONObject.NULL.equals(json
+                                            .get("questionDependency"))) {
+                                QuestionDependencyDto dep = new QuestionDependencyDto();
+                                JSONObject depJson = json
+                                        .getJSONObject("questionDependency");
+                                dep.setQuestionId(depJson
+                                        .getLong("questionId"));
+                                dep.setAnswerValue(depJson
+                                        .getString("answerValue"));
+                                dto.setQuestionDependency(dep);
```

Comment:
@muloem The diff is a bit subtle, but I moved this block of code out of the if statement (on line 958) that starts with
````
if (json.has("optionContainerDto")
````
Otherwise the check for questionDependency is not done for all question types.